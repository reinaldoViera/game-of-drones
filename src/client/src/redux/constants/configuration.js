export const ADD_CONFIG = 'Add new config';
export const ADD_CONFIGS = 'Add configs';
export const UPD_CONFIG = 'Update a config';
export const SERVER_LOADING = 'Loading configs';
export const ERROR_SERVER_LOADING = 'Error loading configs';