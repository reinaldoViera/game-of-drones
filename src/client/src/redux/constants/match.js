export const ADD_MATCH = 'Add new match';
export const ADD_MATCHS = 'Add matchs';
export const UPD_MATCH = 'Update a match';
export const ERROR_SERVER_LOADING = 'Loading matchs';
export const SERVER_LOADING = 'Error Loading matchs';