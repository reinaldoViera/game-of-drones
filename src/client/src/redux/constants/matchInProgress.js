export const START_MATCH = 'Start new match'
export const SERVER_START_MATCH = 'Start new match in server'
export const SERVER_LOADING = 'Loading data from server'
export const ERROR_SERVER_LOADING = 'Error loading data from server'
export const ADD_PLAYER_1 = 'Add player 1'
export const ADD_PLAYER_2 = 'Add player 2'
export const RESET_GAME = 'Reset the match'