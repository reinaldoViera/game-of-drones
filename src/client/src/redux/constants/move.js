export const ADD_MOVE = 'Add new move';
export const ADD_MOVES = 'Add moves';
export const UPD_MOVE = 'Update a move';
export const SERVER_LOADING = 'Loading moves';
export const ERROR_SERVER_LOADING = 'Error loading moves';