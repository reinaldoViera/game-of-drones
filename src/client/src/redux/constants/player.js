export const ADD_PLAYER = 'Add new player';
export const ADD_PLAYERS = 'Add players';
export const UPD_PLAYER = 'Update a player';
export const SERVER_LOADING = 'Loading players';
export const ERROR_SERVER_LOADING = 'Error loading players';