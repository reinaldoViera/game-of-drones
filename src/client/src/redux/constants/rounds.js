export const PLAYER_1_MOVE = 'Player 1 move';
export const PLAYER_2_MOVE = 'Player 2 move';
export const ADD_PREV_ROUND = 'Add prev round';
export const SERVER_LOADING = 'Round in progress';
export const ERROR_SERVER_LOADING = 'Error in round';
export const RESET_GAME = 'Reset the rounds';