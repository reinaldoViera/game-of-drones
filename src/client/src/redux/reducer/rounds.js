import {
    PLAYER_1_MOVE, PLAYER_2_MOVE, ADD_PREV_ROUND, SERVER_LOADING, ERROR_SERVER_LOADING, RESET_GAME
} from "../constants/rounds";


function prevRoundReducer(state = [], { type, payload}) {
    switch (type) {
        case ADD_PREV_ROUND:
            return [
                ...state,
                {
                    winner: payload.winner,
                    name: payload.name,
                    moves: payload.moves
                }
            ]
        default:
            return state
    }
}


const initialState = {
    p1_move: undefined,
    p2_move: undefined,
    in_progress: false,
    error_progress: false,
    winner: false,
    prev_rounds: []
}

export default (state = initialState, {
    type,
    payload
}) => {
    switch (type) {
        case SERVER_LOADING:
            return {
                ...state,
                error_progress: payload ? false : state.error_progress,
                in_progress: payload
            }
        case ERROR_SERVER_LOADING:
            return {
                ...state,
                in_progress: payload
            }
        case PLAYER_1_MOVE:
            return { 
                ...state,
                p1_move: payload
            }
        case PLAYER_2_MOVE:
            return { 
                ...state,
                p2_move: payload
            }
        case ADD_PREV_ROUND:
            return {
                p1_move: undefined,
                winner: payload.matchWinner ? payload.matchWinner : state.winner,
                p2_move: undefined,
                prev_rounds: prevRoundReducer(state.prev_rounds, { type, payload })
            }
        case RESET_GAME:
            return initialState;
        default:
            return state
    }
}