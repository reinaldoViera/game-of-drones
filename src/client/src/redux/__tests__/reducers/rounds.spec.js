import roundReducer from '../../reducer/rounds';
import * as actions from '../../constants/rounds'

describe('Reducer for rounds', () => {
    test('should give initial state', () => {
        const state = roundReducer(undefined, {
            type: 'UNHANDLED'
        });
        expect(state).toMatchSnapshot();
    });
    test('should add moves', () => {
        let state = roundReducer(undefined, {
            type: actions.PLAYER_1_MOVE,
            payload: 1
        });
        expect(state).toMatchSnapshot();
        state = roundReducer(undefined, {
            type: actions.PLAYER_2_MOVE,
            payload: 2
        });
        expect(state).toMatchSnapshot();
    });
    test('should add prev round', () => {
        let state = roundReducer({
            "error_progress": false,
            "in_progress": false,
            "p1_move": 1,
            "p2_move": 2,
            "prev_rounds": [{
                winner: 2,
                name: 'Player 2'
            }],
            "winner": false,
        }, {
            type: actions.ADD_PREV_ROUND,
            payload: {
                winner: 2,
                name: 'Player 1',
                moves: []
            }
        });
        expect(state).toMatchSnapshot();
    })


});