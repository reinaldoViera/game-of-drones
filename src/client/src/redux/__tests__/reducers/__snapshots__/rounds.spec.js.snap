// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer for rounds should add moves 1`] = `
Object {
  "error_progress": false,
  "in_progress": false,
  "p1_move": 1,
  "p2_move": undefined,
  "prev_rounds": Array [],
  "winner": false,
}
`;

exports[`Reducer for rounds should add moves 2`] = `
Object {
  "error_progress": false,
  "in_progress": false,
  "p1_move": undefined,
  "p2_move": 2,
  "prev_rounds": Array [],
  "winner": false,
}
`;

exports[`Reducer for rounds should add prev round 1`] = `
Object {
  "p1_move": undefined,
  "p2_move": undefined,
  "prev_rounds": Array [
    Object {
      "name": "Player 2",
      "winner": 2,
    },
    Object {
      "moves": Array [],
      "name": "Player 1",
      "winner": 2,
    },
  ],
  "winner": false,
}
`;

exports[`Reducer for rounds should give initial state 1`] = `
Object {
  "error_progress": false,
  "in_progress": false,
  "p1_move": undefined,
  "p2_move": undefined,
  "prev_rounds": Array [],
  "winner": false,
}
`;
